README.md

```diff
@@ -201,6 +201,11 @@ Artifact Manager on S3 plugin would try to use the IAM instance profile credenti
 
 ![](images/configue-credentials.png)
 
+Every time you archive/unarchive or download an artifact, Jenkins will generate a temporal URL, it will be valid for an hour, 
+so if you try to reuse an artifact download URL one hour later was generated, it will not be valid, 
+you cannot download the artifact with that URL any more, thus you have to go back to Jenkins 
+and click on the artifact to download it again. 
```

Comment:
Well this discusses download (from a browser?) but not `archive`/`unarchive` (and I suppose `stash`/`unstash` and `copyartifact`). IIUC the significance of the one-hour expiration for these is that a clock starts running at the moment the step starts (which is when the master signs the URLs) and if the agent fails to _initiate_ the upload/download operation within one hour, it will fail. Now if you are only dealing with a single blob (which includes stashes—these are a single tarball), such a delay is extremely unlikely; but if you are, say, `archive`ing a directory full of gigantic files, it could happen that you get halfway through an hour in and then fail to complete the operation.

(`RobustHTTPClient.timeout` defaults to 15m, so this scenario would need to involve at least five file operations, each of which takes several minutes.)

Reply:
yep, this is mostly for user and GUI, for all the rest archive/unarchive/stash/unstash/copyArtifacts the token is created when we call toExternalURL method that I supposed that it is just before call the RobustHTTPClient to make the thing. Is that right?

Reply:
Not really. _All_ the URLs to be passed to the `MasterToSlaveCallable` are created. Then the callable is delivered to the agent and starts running, which you hope is in a matter of milliseconds. Then the agent starts processing them one by one, which could take who knows how long if the blobs involved are big and/or your external network has low bandwidth.